backend/tests/test_bookings.py

```
@@ -0,0 +1,325 @@
+    farmer_repo.create("farmer-a", {})
+    farmer_repo.create("farmer-b", {})
+    payload = SlotBookingCreate(
+        centre_id="ctr-tiny", slot_date=date.today() + timedelta(days=1), slot_window="08:00-10:00"
+    )
+
+    first = slot_service.book_slot(booking_repo, centre_repo, farmer_repo, crop_repo, "farmer-a", payload)
+    assert first.status == "booked"
+
+    try:
+        slot_service.book_slot(booking_repo, centre_repo, farmer_repo, crop_repo, "farmer-b", payload)
+        assert False, "expected ConflictError for a full slot"
```

Ruff (0.16.3)

[warning] Assertion always fails, replace with `pytest.fail()`

(PT015)